src/main/java/hudson/remoting/Channel.java

```diff
@@ -595,6 +598,29 @@ public void terminate(IOException e) {
          *      Otherwise null.
          */
         public void onClosed(Channel channel, IOException cause) {}
+
+        /**
+         * Called when a command is successfully received by a channel.
+         * From outside Remoting, commands are of generally opaque type (beyond being {@link Serializable}),
+         * so listeners can only tally by {@link Object#getClass}.
+         * However {@link Object#toString} will usually be meaningful.
+         * @param channel a channel
+         * @param cmd a command
+         * @param blockSize the number of bytes used to read this command
+         * @since FIXME
+         */
+        public void read(Channel channel, Object cmd, long blockSize) {}
```

Comment:
Should it be onRead/onWrite ?

Reply:
I figured the `on` part was implicit in this being a listener, but I suppose I should change it for consistency with the existing method.